actors/builtin/miner/miner_actor.go

```diff
@@ -1061,15 +1064,33 @@ func (a Actor) ConfirmSectorProofsValid(rt Runtime, params *builtin.ConfirmSecto
 	precommittedSectors, err := st.FindPrecommittedSectors(store, params.Sectors...)
 	builtin.RequireNoErr(rt, err, exitcode.ErrIllegalState, "failed to load pre-committed sectors")
 
-	confirmSectorProofsValid(rt, precommittedSectors)
+	confirmSectorProofsValid(rt, precommittedSectors, params)
 
 	return nil
 }
 
-func confirmSectorProofsValid(rt Runtime, preCommits []*SectorPreCommitOnChainInfo) {
-	// get network stats from other actors
-	rewardStats := requestCurrentEpochBlockReward(rt)
-	pwrTotal := requestCurrentTotalPower(rt)
+func confirmSectorProofsValid(rt Runtime, preCommits []*SectorPreCommitOnChainInfo, params *builtin.ConfirmSectorProofsParams) {
+	var rewardStatsThisEpochBaselinePower big.Int
+	var rewardStatsThisEpochRewardSmoothed smoothing.FilterEstimate
+	var pwrTotalQualityAdjPowerSmoothed smoothing.FilterEstimate
+
+	if params.PrecomputeRewardPowerStats {
```

Comment:
To clean this up you should instead pass the needed reward and power values in as arguments to `confirmSectorProofsValid`.  Then the callers of this subfunction can either pass in values they already have (`ConfirmSectorProofsValid`) or send requests to power and reward actors (`ProveCommitAggregate`)

Reply:
fixed in commit that's incoming....